tests/ranged_balance.cpp

```diff
@@ -0,0 +1,123 @@
+#include "catch/catch.hpp"
+
+#include "dispersion.h"
+#include "game.h"
+#include "monattack.h"
+#include "monster.h"
+#include "npc.h"
+#include "explosion.h"
+#include "projectile.h"
+
+#include "test_statistics.h"
+#include "map_helpers.h"
+
+projectile make_gun_projectile( const item &gun );
+dealt_projectile_attack projectile_attack( const projectile &proj_arg, const tripoint &source,
+                                           const tripoint &target_arg, dispersion_sources dispersion,
+                                           Creature *origin, const vehicle *in_veh );
+
+static void equip_shooter( npc &shooter, itype_id gun_id )
+{
+    tripoint shooter_pos( 60, 60, 0 );
+    shooter.setpos( shooter_pos );
+    shooter.worn.clear();
+    shooter.inv.clear();
+    shooter.remove_weapon();
+    // So we don't drop anything.
+    shooter.wear_item( item( "backpack" ) );
+
+    // Give shooter a loaded gun of the requested type.
+    item &gun = shooter.i_add( item( gun_id ) );
+    const itype_id ammo_id = gun.ammo_default();
+    if( gun.magazine_integral() ) {
+        item &ammo = shooter.i_add( item( ammo_id, calendar::turn, gun.ammo_capacity() ) );
+	REQUIRE( gun.is_reloadable_with( ammo_id ) );
+	REQUIRE( shooter.can_reload( gun, ammo_id ) );
+	gun.reload( shooter, item_location( shooter, &ammo ), gun.ammo_capacity() );
```

Comment:
I'm partially waiting on #21494 and partially just being sloppy since I'm still rewriting large sections of it as I go.  Will be 100% astyled by the time I mark it ready to merge.